packages/store/src/internal/state-factory.ts

```diff
@@ -211,6 +212,28 @@ export class StateFactory {
     return forkJoin(results);
   }
 
+  private checkDuplicateStates(stateClasses: StateClass[]): StateClass[] {
+    const unique: StateClass[] = [];
+    const occurrences: Occurrences = {};
+
+    stateClasses
+      .filter((state, position, list) => list.indexOf(state) === position)
+      .forEach(state => {
+        const meta: Partial<MetaDataModel> = state[META_KEY] || {};
+        const stateName: string = String(meta.name);
+        const compareState = occurrences[stateName];
+
+        if (compareState) {
+          console.warn(`State name in ${state.name} already exists also in`, compareState);
```

Comment:
```suggestion
          console.warn(`State name `${stateName}` in ${state.name} already exists in`, compareState);
```
Slight improvement in message providing the state name.